src/library/scala/collection/StringParsers.scala

```diff
@@ -246,13 +246,13 @@ final private[scala] object StringParsers {
       //but not just .
       val startChar = format.charAt(startIndex)
       if(startChar == '.') {
-         val noSignificant = skipIndexWhile(ch => ch >= '0' && ch <= '9', startIndex + 1, endIndex)
-          if(noSignificant == endIndex) (noSignificant != startIndex + 1) //not just "."
-          else {
-            val e = format.charAt(noSignificant)
-            if (e == 'e' || e == 'E') expOK(noSignificant + 1, endIndex)
-            else false
-          }
+        val noSignificant = skipIndexWhile(ch => ch >= '0' && ch <= '9', startIndex + 1, endIndex)
+        if(noSignificant == startIndex + 1) false //not just "." or ".Exxx"
+        else {
+          val e = format.charAt(noSignificant)
+          if (e == 'e' || e == 'E') expOK(noSignificant + 1, endIndex)
+          else false
+        }
```

Comment:
Speaking of formatting, no one mentioned the variant space after `if`. This code base is mostly but not absolutely for the space, which is my preference; but a couple of people omit it. Probably it's best to pick one and gird for battle.

Reply:
I tend to be fully consistent with the rule that I use, like, whatever feels right at the time, man.

But considering I've annoyed you several times in the last week already, and I don't want to stay consistent in that regard, I'll put in some spaces.

Reply:
i.e. `(e == 'e' || e == 'E') && expOK(...)`

Reply:
I read that we're going for parseDouble compatibility. Just to mention that Scala doesn't take `5.E2` as a constant anymore, because of the ambiguity with selection.

Reply:
And then: 
`noSignificant != startIndex + 1 && {
  val e = ???
 (e == 'e' || ...) && expOK(...)
}

Reply:
@som-snytt  This entire validation monster returns a boolean and is mostly made out of expressions that return booleans, so I could make it all in to a vary long series of nested and chained `&&` and `||` 's. This fragment could be 

```
(startChar == '.' && {
  val noSignificant = skipIndexWhile(ch => ch >= '0' && ch <= '9', startIndex + 1, endIndex)
  (noSignificant != startIndex +1)
} && {
  val e = format.charAt(noSignificant)
  (e == 'e' || e == 'E') && expOP(noSignificant + 1, endIndex)
} || (...)
```

But the explicit if/elses with boolean literal returns in some places make it *much* easier to follow for me here. There probably is some arbitrary line where opinions can differ. I have no strong feelings where that line should be, so if there are particular places where you'd like to see it moved, I'm fine with that.

Reply:
Yes, there is a line of taste. Returning true might be a code smell, but I trust your judgment.